### DISCUSSION TITLE
Support for Multiple Projects in a Single Workspace

### DISCUSSION THREAD

**Discussion by lucaschultz** [biomejs/biome-vscode#426] [Q&A]
Hi,

I saw in [this PR](https://github.com/biomejs/website/pull/974) that support for multiple projects in one workspace is coming soon. I followed the instructions in the documentation by adding

```json
"biome.projects": [
  {
    "path": "apps/reference-api"
  }
]
```

to my `.vscode/settings.json` and the mono repo root, and I installed the pre-release version of the extension, but it didn't work. My folder structure is:

```
.
├── .vscode
│   └── settings.json
├── apps
│   └── reference-api
│       ├── biome.jsonc
│       ├── package.json
│       ├── src
│       └── tsconfig.json
├── biome.jsonc
├── package.json
├── packages
│   └── config
│       └── tsconfig
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
└── turbo.json
```

In `apps/reference-api/biome.jsonc`, I extend from `biome.jsonc` but disable a rule:

```json
{
  "$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
  "extends": ["../../biome.jsonc"],
  "linter": {
    "rules": {
      "suspicious": {
        "noExplicitAny": {
          "level": "off"
        }
      }
    }
  }
}
```

This setup works with the CLI, but not with the VSCode extension. Am I doing something wrong, or is this feature not yet supported in the pre-release version of the VSCode extension?

**Comment by nhedger** [comment]
Hey,

I would expect this to work.

Are you not getting the diagnostics at all, or is it that the config file is not being used ?

Anyway, would you mind creating a reproduction repo so we can try and reproduce ? You can use `npm create @biomejs/biome-reproduction@latest`

**Comment by chiragmRoado** [comment]
@nhedger Similar issue with my setup as well.

Here's the setup overview:
```
"biome.projects": [
  {
    "path": "app1"
  },
  {
    "path": "app2"
   }
]

```
```
.
├── biome.json (Root in Working Directory)
├── app1
│       ├── biome.json
│       ├── package.json
│       ├── src
│       └── tsconfig.json
├── app2
│       ├── biome.json
│       ├── package.json
│       ├── src
│       └── tsconfig.json
├── app3
│       ├── biome.json
│       ├── package.json
│       ├── src
│       └── tsconfig.json
```

Each application is structured as a separate repository, with its own biome.json configuration file to handle unique linting and formatting rules.

Biome plugin does not appear to recognize or apply the individual configurations for each repository correctly. Instead, it crashes unless a single biome.json is placed at the root level. While this workaround enables the plugin to function, it disregards the specific rules defined in each repository, which is not ideal.

The only alternative right now is to open each project in a separate window, but that's not ideal given how our projects are structured.

**Reply by ematipico** [reply]
As mentioned here https://github.com/biomejs/biome-vscode/discussions/426#discussioncomment-11278835, we would need a reproduction. A minimal one.

**Comment by ekoeditaa** [comment]
@ematipico I have a sample reproduction here: https://github.com/ekoeditaa/biome-repro-1736738729229. 
Running `yarn lint` in root shows warning because in root `biome.json` we have set `"useNodejsImportProtocol": "warn"`
```
Checked 23 files in 2ms. No fixes applied.
$ biome lint .
Found 4 warnings.
✨  Done in 0.23s.
``` 

Running `yarn lint` in `app_one` shows no warning as we have disabled the rule in `app_one/biome.json`:

```
$ biome lint .
Checked 11 files in 1907µs. No fixes applied.
✨  Done in 0.22s.
```

When opening the file `app_one/utils.ts` I would expect to not get this lint error as I have disabled the rule in `app_one/biome.json`. However, this is what I get:
<img width="971" alt="image" src="https://github.com/user-attachments/assets/821ba562-2cc5-4e15-812a-eff1746e191b" />

**Comment by lasseklovstad** [comment]
I tried the pre-release version aswell and i couldn't get it to work with settings.biome.projects. Any updates on this? I realy don't want to have to be forced to use vscode workspaces to be able to use the biome extension in my monorepo 🥲
